Packs/CommonScripts/Scripts/TextFromHTML/TextFromHTML.py

```diff
@@ -5,26 +5,57 @@
 import re
 
 
-def get_plain_text(html_regex):
+def get_plain_text(html: str):
     data = ''
-    if html_regex and html_regex.group(0):
-        data = re.sub(r'<.*?>', '', html_regex.group(0))
+    if html:
+        data = re.sub(r'<br>', '\n', html, flags=re.M + re.S + re.I + re.U)
+        data = re.sub(r'<.*?>', '', html, flags=re.M + re.S + re.I + re.U)
         entities = {'quot': '"', 'amp': '&', 'apos': "'", 'lt': '<', 'gt': '>', 'nbsp': ' ',
                     'copy': '(C)', 'reg': '(R)', 'tilde': '~', 'ldquo': '"', 'rdquo': '"', 'hellip': '...'}
         for e in entities:
             data = data.replace(f'&{e};', entities[e])
+        data = re.sub(r'[ \t]{2,}', ' ', data)
+        data = re.sub(r'(\r?\n){3,}', '\n\n', data)
+        data = re.sub(r'\s+$', '', data)
     return data
 
 
-def text_from_html(args):
-    html = args['html']
+def get_body(html: str, html_tag: str, allow_fallback: bool = False):
+    if html and html_tag:
+        body = re.search(fr'<{html_tag}.*/{html_tag}>', html, re.M + re.S + re.I + re.U)
+
+        if body and body.group(0):
+            return body.group(0)
+        elif allow_fallback and html_tag.lower() == 'body':
+            return html
+    return ''
+
+
+def text_from_html(args: dict):
+    html = args.get('html', '')
     html_tag = args.get('html_tag', 'body')
+    allow_fallback = str(args.get('allow_body_fallback', 'false')).lower() == 'true'
 
-    body = re.search(fr'<{html_tag}.*/{html_tag}>', html, re.M + re.S + re.I + re.U)
+    body = get_body(html, html_tag, allow_fallback)
     data = get_plain_text(body)
+
     return data if data != '' else 'Could not extract text'
 
 
+def main():
+    try:
+        text = text_from_html(demisto.args())
+        context_path = demisto.args().get('context_path', None)
+        result = CommandResults(
+            outputs_prefix=context_path,
+            outputs=text if context_path else None,
```

Comment:
I've made it a constant "TextFromHTML" context key, but kept it optional if the context key is created or not.
We could make it always the case, but for the sake of backwords-compability and to avoid blowing up the context of existing playbooks, if the feature isn't required, i made it optional.
 For people just using this in the warroom, putting it to the context may not be what they want.
 Not sure, what the best practise here would be.
 Would it be better to always have the context key, but only populate it if needed?

Reply:
@MichaelYochpaz Can you let me know, if this is OK now, or what you would recommend?

Reply:
Adding additional stuff to context is not considered breaking backwards-compitability (removing / changing keys does), but if you think there could be cases where users might not want to have it added to context, adding an optional parameter to determine whether to add it to context sounds fine to me.
Up to you.

Reply:
I'm sticking with making it optional.